### ISSUE TITLE
Myst-generated header anchors do not work

### ISSUE THREAD

**Issue by mason3k** [sphinx-contrib/confluencebuilder#846]
## Issue
It appears to me that auto-generated header anchor links are not working. After some investigation, it seems to be because they are generating as the lower-case dash-separated header name, whereas Confluence anchor links are case sensitive ([per this article](https://support.atlassian.com/confluence-cloud/docs/insert-links-and-anchors/)).

So for instance, when running the builder, the link would generate as:
[confluence url]/\~[space key]/pages/[page_id]/PAGE+NAME#version-history
whereas a link natively generated within the editor would be: 
[confluence url]/\~[space key]/pages/[page_id]/PAGE+NAME#**Version-History**

As a result, the links do not work. 

## Other Details
Confluence V2 Editor
Confluence Builder v 2.2.0
I'm generating docs with markdown/myst and [auto-generated header anchors](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-header-anchors).

**Comment by mason3k** [comment]
I don't know if this is the appropriate fix, but in `translator.py`, function `visit_title`, the links are generated correctly if I change

```python
elif 'refid' in node:
    target = '-'.join(node.astext().split()).lower()
    ...
```

to

```python
elif 'refid' in node:
    target = '-'.join(node.astext().split())
    ...
```

**Comment by jdknight** [comment]
@mason3k, you happen to have a small sample set for this? I recently been trying to create a small example to show this issue so I can include an appropriate unit test along with a fix. While I do see the issue with V2 linking to anchors on other pages (which I have a different fix for), I haven't been able to exercise the line you and referenced yet (although, I'll keep trying).

**Comment by mason3k** [comment]
Sorry for the delay! I played with this and have some more information:

I don't actually believe the issue has to do with the code above. It appears to me that explicit header anchors _are_ working, but [myst auto-generated headers](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#auto-generated-header-anchors) are not. This setting allows Myst users to use the typical markdown convention for linking to headers, e.g., `[](#header-anchor)` for a section in the same doc titled Header Anchor. 

These references would work out of the box if Confluence didn't have the (IMO kind of weird) case-sensitive header id convention. But as it is, to support this we would probably need to create the lowercase id's as anchors when we visit the headers so that the links are valid. The other, more convoluted option would be to store the target id as case-sensitive when we visit the reference node.

Here is some code that works on my machine:
```python
    def visit_section(self, node):
        level = self._section_level

        ...
        
        # slug is added by the Myst parser to nodes that should have an anchor generated
        if 'slug' in node:
            node['section-embedded-anchor'] = node['slug']
```
`sphinxcontrib/confluencebuilder/translator.py`

`node['slug'] = <header slug>` is added by the Myst parser for sections that should be linkable and, as far as know, is not used otherwise. `section-embedded-anchor` is referenced later when visiting titles to indicate that the section should have an anchor generated. The end result is that these headers have anchors with lowercase slugs, but only for Myst users who are autogenerating headers. 

P.S. Since this is an integration with an alternate parser, I think it would be fair (but kind of sad 😊) to not support this.

**Comment by jdknight** [comment]
Thanks for the additional information. I will take another look at it, utilizing `[](#header-anchor)` like markup to test.

All for supporting the MyST-Parser here. It's a great extension that works nicely with Sphinx ~ and we try our best to get it to work with this Confluence extension as much as we can.

**Comment by jdknight** [comment]
Changes have been introduced (#866) to help rework how anchors are generated and linked to across various pages. This extension should now have better support when linking across different pages, no matter if its a v1 editor page or a v2 editor page. Note that Confluence anchor links sometimes may take a long time to jump to, or maybe offset near a target (unknown on if/when Atlassian plans to improve this user experience).

These changes will be made available next release (v2.4), and can be tested now by using the [development version](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation). For now, it will be assumed that this issue has been resolved, and will tag this issue appropriately. Feel free to try it out ahead of time any report any issues you may have.

**Comment by jdknight** [comment]
v2.4 is now available on PyPI -- marking as closed.
